app/api/tax.py

```diff
@@ -43,19 +45,20 @@ class TaxList(ResourceList):
 
     def query(self, view_kwargs):
         query_ = self.session.query(Tax)
-        if view_kwargs.get('id'):
-            query_ = query_.join(Event).filter(Event.id == view_kwargs['id'])
+        if view_kwargs.get('event_id'):
+            query_ = query_.join(Event).filter(Event.id == view_kwargs['event_id'])
         elif view_kwargs.get('identifier'):
             query_ = query_.join(Event).filter(Event.identifier == view_kwargs['identifier'])
         return query_
 
     def before_create_object(self, data, view_kwargs):
-        if view_kwargs.get('id'):
-            event = self.session.query(Event).filter_by(id=view_kwargs['id']).one()
+        if view_kwargs.get('event_id'):
+            event = self.session.query(Event).filter_by(id=view_kwargs['event_id']).one()
```

Comment:
I think it is good to add `ObjectNotFound` check here. That's true that I won't be able to store the record since ForeignKey will not be matched for `event_id` and even before that `.one()` will throw error but I think it will be 500 error instead its should be  404 or unprocessable